include/ibamr/INSVCStaggeredConservativeConvectiveOperator.h

```diff
@@ -196,6 +197,29 @@ class INSVCStaggeredConservativeConvectiveOperator : public ConvectiveOperator
      */
     void setMassDensitySourceTerm(const SAMRAI::tbox::Pointer<IBTK::CartGridFunction> S_fcn);
 
+    /*!
+     * \brief Set the patch data indices corresponding to the velocity at the previous time step
+     * to be used when computing the density update
+     *
+     * \note This velocities will be used to compute an approximation to velocities required for SSPRK updates
+     * V_old_idx = n-1, V_current_idx = n, V_new_idx = n+1,k (after an INS cycle)
+     * If V_old_idx or V_new_idx are not set, then they will degenerate to V_current automatically, for the very first
+     * simulation time step and cases where an INS cycle has not been executed, respectively.
+     */
+    void setFluidVelocityPatchDataIndices(int V_old_idx, int V_current_idx, int V_new_idx);
+
+    /*!
+     * \brief Set the cycle number currently being executed by the INS integrator. This will determine the rho advection
+     * velocity.
+     */
+    void setCycleNumber(int cycle_num);
```

Comment:
Can this be removed in lieu of the approach used in the other convective operators?